### PR TITLE
Fixes from demonstrations in January 2023

### PR DESCRIPTION
**Pull request by ksnavely**
- [x] Tick to sign-off your agreement to the [Developer Certificate of Origin (DCO) 1.1](../blob/master/DCO1.1.txt)

## What

In this pull request I seek to make the demo runnable again.

1. The World Clock API is depreciated. Use a GitHub API which can be accessed easily and should be fairly stable.
2. I suspect binary evidence support was broken when agent-mode was added. I tweaked a conditional to restore support for signed binary evidence.

## Why

To better evangelize Auditree I want the demo working well.

## How

- Update the world clock example to use the GitHub versions endpoint
- Don't attempt to encode already-bytes data

## Test

The demo is working, both the fetch step and the check step.
Unit tests are passing.

## Context

Try the demo off main today to see the failures.
Linting and tests pass locally.

### CODE REVIEW

**Comment by ksnavely** [comment]
I'm accepting comments on this PR. I think a new test covering binary signed evidence would be a last change worth including -- it should fail off `main` today and pass on my branch.

**Comment by drsm79** [comment]
Thanks for this, the change looks good from a quick eyeball. 
One thing, calling out to the API was a nice part of the demo, as it shows the whole collect remote responses. 
Obviously that needs to be a working API, maybe calling a GitHub api/URL for this project would be a good additional demo?

File: compliance/evidence.py [hunk 1]
```diff
@@ -224,8 +226,12 @@ def set_content(self, content, sign=True):
         if self.extension == 'json':
             self._content = format_json(json.loads(content))
         if sign:
+            if self.binary_content:
+                data_bytes = self._content
+            else:
+                data_bytes = self._content.encode()
             self._digest, self._signature = self.agent.hash_and_sign(
-                self._content.encode()
+                data_bytes
```
Comment (cletomartin) [inline compliance/evidence.py:234]: Nice. This will support now signing binary data.

**Review by cletomartin — COMMENTED** [review]
All looks good. You will need to update the version (I think a minor one) and specify the changes in CHANGES.md.

**Comment by ksnavely** [comment]
@drsm79 I've added a GitHub API example -- what do you think of using the versions API?

**Review by drsm79 — APPROVED** [review]

**Comment by drsm79** [comment]
> @drsm79 I've added a GitHub API example -- what do you think of using the versions API?

Nice, good choice.

**Comment by ksnavely** [comment]
I've added a binary signing test that now passes but fails on main.

**Review by cletomartin — APPROVED** [review]

**Comment by cletomartin** [comment]
@ksnavely perhaps we should merge #144 this first as it will fix some problems with the current CI config.

**Comment by ksnavely** [comment]
I should be good to go after you merge your PR, I can't recall if I'll need to re-push this branch after an update or not.

**Comment by cletomartin** [comment]
Yeah, there is some conflicts in there after my merge of #144

**Review by cletomartin — APPROVED** [review]

**Pull request merged**